src/ItemFieldSet.tsx

```diff
@@ -66,9 +117,85 @@ export class ItemFieldSet extends React.Component<IItemFieldSetProps, IItemField
     this.setState({ items });
   }
 
+  private changeItemPrice = (index: number) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const items: IItem[] = [...this.state.items];
+    const newPrice = event.target.value.replace(/[^0-9.]/, '');
+    items[index].price = newPrice;
+    this.setState({ items });
+    this.props.setItems(this.state.items);
+  }
+
   private changeItemQuantity = (index: number) => (event: React.ChangeEvent<HTMLInputElement>) => {
     const items: IItem[] = [...this.state.items];
-    items[index].quantity = parseInt(event.target.value, 10);
+    const newQuantity = event.target.value.replace(/[^0-9]/, '');
+    if (!this.isNumber(newQuantity)) {
+      return
+    }
+    items[index].quantity = newQuantity;
+    this.setState({ items });
+    this.props.setItems(this.state.items);
+  }
+
+  private itemPriceToggleFocus = (index: number, focused: boolean) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const items: IItem[] = [...this.state.items];
+    items[index].isFocused = focused;
```

Comment:
This is problematic, because items[index] is not being modified and not created a new object. Do this instead:
items[index] = { ...items[index], is focused: true };